README.md

```diff
@@ -1,6 +1,16 @@
 # Mbed TLS Examples on Mbed OS
 
-This repository contains a collection of Mbed TLS example applications based on Mbed OS. Each subdirectory contains a separate example meant for building as an executable.
+This repository contains following Mbed TLS example applications based on Mbed OS. Each subdirectory contains a separate example meant for building as an executable. Please browse to the subdirectories for specific documentation.
+
+* [authcrypt](./authcrypt/README.md): performs authenticated encryption and authenticated decryption of a buffer.
+* [benchmark](./benchmark/README.md): benchmarks the various cryptographic primitives offered by Mbed TLS.
+* [hashing](./hashing/README.md): performs hashing of a buffer with SHA-256 using various APIs.
+* [tls-client](./tls-client/README.md): downloads a file from an HTTPS server (os.mbed.com) and looks for a specific string in that file.
+
+In addition following examples demonstrate the Mbed TLS opaque keys feature using hardware crypto engine [ATECC508A](https://github.com/ARMmbed/mbed-os/tree/feature-opaque-keys/features/atcryptoauth/README.md).
+* [pk_sign](./pk_sign/README.md): create signature using opaque keys and verify in software.
+* [pk_verify](./pk_verify/README.md): create and verify signature using opaque keys.
+* [tls-client-with-client-authentication](./tls-client-with-client-authentication/README.md): similar to the tls-client example. Except it demonstrates client authentication and opaque keys. 
```

Comment:
Grammar: you can't put a full stop before “except”. But it would be better to make the example stand on its own. Also this client does not work against `os.mbed.com` (because that wouldn't use client authentication). Suggestion:

> downloads a file from an HTTPS server with client authentication. The client authentication key can be an opaque key.